extensions/amp-ad-exit/0.1/filters/click-location.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {FilterType, Filter} from './filter';
+import {user} from '../../../../src/log';
+
+export class ClickLocationFilter extends Filter {
+  /**
+   * @param {string} name The user-defined name of the filter.
+   * @param {!../config.ClickLocationConfig} spec
+   * @param {!../AmpAdExit} adExitElement
+   */
+  constructor(name, spec, adExitElement) {
+    super(name);
+    user().assert(isValidClickLocationSpec(spec), 'Invaid ClickLocation spec');
+
+    /** @private {number} */
+    this.leftBorder_ = spec.left || 0;
+
+    /** @private {number} */
+    this.rightBorder_ = spec.right || 0;
+
+    /** @private {number} */
+    this.topBorder_ = spec.top || 0;
+
+    /** @private {number} */
+    this.bottomBorder_ = spec.bottom || 0;
+
+    /**
+     * By default, the border protection is relative to the current iframe. User
+     * can specify a relativeTo element, then the border protection is relative
+     * to the bounding rect of that element.
+     * @private {string|undefined}
+     */
+    this.relativeTo_ = spec.relativeTo;
+
+    /** @private {!../AmpAdExit} */
+    this.adExitElement_ = adExitElement;
+
+    /**
+     * The structure that represents the rect that will not be filtered .
+     *
+     * @typedef {{
+     *   top: number,
+     *   right: number,
+     *   bottom: number,
+     *   left: number
+     * }}
+     */
+    this.allowedRect_ = {};
+  }
+
+  /** @override */
+  filter(event) {
+    if (event.clientX >= this.allowedRect_.left &&
```

Comment:
I think we can leave it for now. It is rare case that the this.allowedRect_ is not ready until click. And since we filter the click within 1 second by default, 1 second should be enough for that to be ready. 
Even if this.allowedRect_ is not ready, the click is just rejected, no side effect will happen.